server/sonar-server/src/main/java/org/sonar/server/qualityprofile/DefinedQProfileInsertImpl.java

```diff
@@ -0,0 +1,162 @@
+  private final TypeValidations typeValidations;
+  private RegisterQualityProfiles.RuleRepository ruleRepository;
+
+  public DefinedQProfileInsertImpl(DbClient dbClient, System2 system2, UuidFactory uuidFactory, TypeValidations typeValidations) {
+    this.dbClient = dbClient;
+    this.system2 = system2;
+    this.uuidFactory = uuidFactory;
+    this.typeValidations = typeValidations;
+  }
+
+  @Override
+  public void create(DbSession session, DefinedQProfile definedQProfile, OrganizationDto organization, List<ActiveRuleChange> changes) {
+    initRuleRepository(session);
+
+    checkArgument(definedQProfile.getParentQProfileName() == null,
+      "Inheritance of Quality Profiles is not supported yet");
+    // Optional.ofNullable(definedQProfile.getParentQProfileName())
+    // .map(parentQProfileName -> dbClient.qualityProfileDao().selectByNameAndLanguage(organization, parentQProfileName.getName(),
+    // parentQProfileName.getLanguage(), session))
+    // .map(parentQualityProfileDto -> dbClient.activeRuleDao().selectByRuleIds());
+
+    Date now = new Date(system2.now());
+    QualityProfileDto profileDto = insertQualityProfile(session, definedQProfile, organization, now);
+
+    definedQProfile.getActiveRules()
+      .stream()
+      .map(activeRule -> insertActiveRule(session, profileDto, activeRule, now.getTime()))
+      .forEach(changes::add);
+
+    insertTemplate(session, definedQProfile, organization);
+  }
+
+  private void initRuleRepository(DbSession session) {
+    if (ruleRepository == null) {
+      ruleRepository = new RegisterQualityProfiles.RuleRepository(dbClient, session);
+    }
+  }
+
+  private QualityProfileDto insertQualityProfile(DbSession session, DefinedQProfile definedQProfile, OrganizationDto organization, Date now) {
+    QualityProfileDto profileDto = QualityProfileDto.createFor(uuidFactory.create())
+      .setName(definedQProfile.getName())
+      .setOrganizationUuid(organization.getUuid())
+      .setLanguage(definedQProfile.getLanguage())
+      .setDefault(definedQProfile.isDefault())
+      .setRulesUpdatedAtAsDate(now);
+    dbClient.qualityProfileDao().insert(session, profileDto);
+    return profileDto;
+  }
+
+  private ActiveRuleChange insertActiveRule(DbSession session, QualityProfileDto profileDto, org.sonar.api.rules.ActiveRule activeRule, long now) {
+    RuleKey ruleKey = RuleKey.of(activeRule.getRepositoryKey(), activeRule.getRuleKey());
+    RuleDefinitionDto ruleDefinitionDto = ruleRepository.getDefinition(ruleKey)
+      .orElseThrow(() -> new IllegalStateException("RuleDefinition not found for key " + ruleKey));
+
+    ActiveRuleDto dto = ActiveRuleDto.createFor(profileDto, ruleDefinitionDto);
+    dto.setSeverity(
+      MoreObjects.firstNonNull(
+        activeRule.getSeverity().name(),
+        // context.parentSeverity(),
+        ruleDefinitionDto.getSeverityString()));
+    // ActiveRule.Inheritance inheritance = activeRule.getInheritance();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACommentedOutCodeLine)